### PR TITLE
fix: Usability Updates Sept 2021

### PR DESCRIPTION
**Pull request by abest0**
*Issue #, if available:*
- Custom Resource missing to manually deploy `MediaLive` and `MediaPackage` resources
- Segment thumbnails appear full size in the `**SidebarTable**` component
- No configurations possible for using custom named stacks with CI/CD

*Description of changes:*
This PR contains a number of usability improvements to address issues introduced by updates to Amplify components. Additionally, modules required for CloudFormation custom resource deployment (removed by `.gitignore`) have been re-added.

### Custom Resources
Previously, messages used to inform Cloudformation of custom resource deployment state were crafted manually. We have updated the backend elemental custom resource to use the [crhelper library](https://github.com/aws-cloudformation/custom-resource-helper) to manage messaging of the custom resource deployment state.

### Segment thumbnails
In previous updates, the amplify libraries and mechanism to import the library components were updated. This set of changes updates the [custom CSS properties](https://docs.amplify.aws/ui/storage/s3-image/q/framework/vue/#props-css-amplify-s3-image) applied to the `<amplify-s3-image>` component to pass width attributes to the component. To further improve the process of displaying segment thumbnails, an additional Graphql resolver has been added for the `Thumbnail_Key` attribute of the `NewSegmentSummary` type path to the resized segment image.

### Custom Named Stacks for CI
The `broadcast-monitoring-ui/setup_config.py` file has been updated to facilitate both building with CI and local development. The UI setup script still accepts the name of a stack as an argument for local development. For CI, the script will default to using `broadcast-monitoring` when a  **`BACKEND_STACK`** environment variable is not defined.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: broadcast-monitoring-ui/src/components/DetailTable.vue [hunk 1]
```diff
@@ -94,6 +94,7 @@
 </template>
 
 <script>
+// import { Storage } from 'aws-amplify'
```
Comment (jonslo) [inline broadcast-monitoring-ui/src/components/DetailTable.vue:97]: Can you remove this commented-out code?

File: broadcast-monitoring-ui/src/components/DetailTable.vue [hunk 2]
```diff
@@ -126,18 +127,18 @@ export default {
...
-        newFrameDetail.Logo_Check &&
-        newFrameDetail.Logo_Check.Detected_Logo_Crop_S3_Key
-      ) {
-        this.cropLogoSrc = await this.$Amplify.Storage.get(
-          newFrameDetail.Logo_Check.Detected_Logo_Crop_S3_Key
-        )
-        return
-      }
-      this.cropLogoSrc = null
+    frameDetail: async newFrameDetail => {
```
Comment (jonslo) [inline broadcast-monitoring-ui/src/components/DetailTable.vue:130]: Can you remove this commented-out code?

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/medialive.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/python
```
Comment (jonslo) [inline broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/medialive.py:1]: Do we need ASL for this if repository is MIT-0?

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/mediapackage.py [hunk 4]
```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/python
```
Comment (jonslo) [inline broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/mediapackage.py:1]: Do we need ASL for this if repository is MIT-0?

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/requirements.txt [hunk 5]
```diff
@@ -0,0 +1 @@
+crhelper
```
Comment (jonslo) [inline broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/requirements.txt:1]: Might be useful to include version of **crhelper**

**Review by jonslo — CHANGES_REQUESTED** [review]

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/medialive.py (hunk 3 above)
Comment (abest0) [reply broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/medialive.py:1]: Good question. Will remove.

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/mediapackage.py (hunk 4 above)
Comment (abest0) [reply broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/mediapackage.py:1]: ASL already exists in a number of files... leaving this as is.

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/requirements.txt (hunk 5 above)
Comment (abest0) [reply broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/requirements.txt]: updated

File: broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/medialive.py (hunk 3 above)
Comment (abest0) [reply broadcast-monitoring/infrastructure/elemental/custom_resources/custom-resource-py/libs/medialive.py:1]: ASL already exists in a number of files... leaving this as is.

**Review by jonslo — APPROVED** [review]

**Pull request merged**